tests-integration/src/tests/reconcile_table.rs

```diff
@@ -0,0 +1,581 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use client::{OutputData, DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME};
+use common_meta::reconciliation::manager::ReconciliationManagerRef;
+use common_meta::reconciliation::ResolveStrategy;
+use common_procedure::ProcedureManagerRef;
+use common_recordbatch::util::collect_batches;
+use common_test_util::recordbatch::check_output_stream;
+use table::table_reference::TableReference;
+
+use crate::cluster::GreptimeDbClusterBuilder;
+use crate::tests::test_util::{
+    dump_kvbackend, execute_sql, restore_kvbackend, try_execute_sql, wait_procedure, MockInstance,
+    MockInstanceBuilder, RebuildableMockInstance, TestContext,
+};
+
+const CREATE_MONITOR_TABLE_SQL: &str = r#"
+CREATE TABLE monitor (
+    t TIMESTAMP TIME INDEX,
+    env STRING INVERTED INDEX,
+    cloud_provider STRING,
+    latency Float,
+    PRIMARY KEY (env, cloud_provider)
+)
+PARTITION ON COLUMNS (`env`) (
+    `env` < 'env-0',
+    `env` >= 'env-0' AND `env` < 'env-1',
+    `env` >= 'env-1' AND `env` < 'env-2',
+    `env` >= 'env-2'
+)
+with('append_mode'='true');"#;
+
+const CREATE_TABLE_SQL: &str = r#"
+CREATE TABLE grpc_latencies (
+    ts TIMESTAMP TIME INDEX,
+    host STRING INVERTED INDEX,
+    method_name STRING,
+    env STRING NOT NULL,
+    latency Float,
+    PRIMARY KEY (host, method_name)
+)
+PARTITION ON COLUMNS (`host`) (
+    `host` < 'host1',
+    `host` >= 'host1' AND `host` < 'host2',
+    `host` >= 'host2' AND `host` < 'host3',
+    `host` >= 'host3'
+)
+with('append_mode'='true');"#;
+
+const INSERT_DATA_SQL: &str = r#"
+INSERT INTO grpc_latencies (
+    ts,
+    host,
+    method_name,
+    latency,
+    env
+) VALUES (
+    '2025-08-08 20:00:06',
+    'host1',
+    'GetUser',
+    103.0,
+    'prod'
+);
+"#;
+
+const INSERT_DATA_SQL_WITHOUT_ENV: &str = r#"
+INSERT INTO grpc_latencies (
+    ts,
+    host,
+    method_name,
+    latency
+) VALUES (
+    '2025-08-08 20:00:07',
+    'host2',
+    'GetUser',
+    104.0
+);
+"#;
+
+const RENAME_TABLE_SQL: &str = r#"
+ALTER TABLE grpc_latencies RENAME grpc_latencies_renamed;
+"#;
+
+const MODIFY_COLUMN_LATENCY_TYPE_SQL: &str = r#"
+ALTER TABLE grpc_latencies MODIFY COLUMN latency DOUBLE;
+"#;
+
+const INSERT_DATA_SQL_WITH_CLOUD_PROVIDER: &str = r#"
+INSERT INTO grpc_latencies (
+    ts,
+    host,
+    method_name,
+    cloud_provider,
+    latency,
+) VALUES (
+    '2025-08-08 20:00:08',
+    'host3',
+    'GetUser',
+    'aws',
+    105.0
+);
+"#;
+
+const ADD_COLUMN_CLOUD_PROVIDER_SQL: &str = r#"
+ALTER TABLE grpc_latencies ADD COLUMN cloud_provider STRING;
+"#;
+
+const ADD_COLUMN_ENV_SQL: &str = r#"
+ALTER TABLE grpc_latencies ADD COLUMN env STRING;
+"#;
+
+const DROP_COLUMN_ENV_SQL: &str = r#"
+ALTER TABLE grpc_latencies DROP COLUMN env;
+"#;
+
+#[tokio::test]
+async fn test_reconcile_dropped_column() {
+    let builder = GreptimeDbClusterBuilder::new("test_reconcile_dropped_column").await;
```

Comment:
You can always `common_telemetry::init_default_ut_logging();` to enable debug log for tests.